activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -104,12 +104,22 @@ def index_name_for_remove(table_name, options = {})
 
       class V5_0 < V5_1
         def create_table(table_name, options = {})
-          if ActiveRecord::Base.connection.adapter_name == "PostgreSQL"
+          connection_name = self.connection.adapter_name
+          if connection_name == "PostgreSQL"
             if options[:id] == :uuid && !options[:default]
               options[:default] = "uuid_generate_v4()"
             end
           end
-          super
+
+          # Since 5.1 Postgres adapter uses bigserial type for primary
+          # keys by default and MySQL uses bigint. This compat layer makes old migrations utilize
+          # serial/int type instead -- the way it used to work before 5.1.
+          if options[:id].blank?
+            options[:id] = (connection_name == "PostgreSQL") ? :serial : :integer
```

Comment:
Can we improve things by making this:

```ruby
options[:id] = :integer
options[:auto_increment] = true
```

... and then leave the SQlite and PostgreSQL adapters to turn that back into whatever internal type they need?